doc-tools/missing-doclet/src/main/java/org/opensearch/missingdoclet/MissingDoclet.java

```diff
@@ -59,12 +60,16 @@ public class MissingDoclet extends StandardDoclet {
     private static final int METHOD = 2;
     // checks that @param tags are present for any method/constructor parameters
     private static final int PARAMETER = 3;
+    // TODO Should we stick this list in missing-javadoc.gradle instead?
+    private static final Set<String> ALLOWED_TO_BE_IGNORED = new HashSet<>(List.of(
```

Comment:
@austintlee I think we could ignore all generated classes by checking the presence of the `@Generated` annotation on the element, the `protoc` supports that:

```
protobuf {
  protoc {
    artifact = "com.google.protobuf:protoc:${versions.protobuf}"
  }
  generateProtoTasks {
    all().configureEach { task ->
      builtins {
        java {
          option "annotate_code"
        }
      }
    }
  }
}
```

Would end up with:


```
@javax.annotation.Generated(value="protoc", comments="annotations:ExtensionIdentityProto.java.pb.meta")
public final class ExtensionIdentityProto {
}
```

Reply:
@reta please take a look at the new diff.

I added support for this annotation.  To do that, I had to add a new dependency to "server" because of "javax.annotation.Generated".  This, however, does not annotate interfaces, e.g. ExtensionRequestOrBuilder so we still need the code for recognizing "*".

Reply:
```
Could not determine the dependencies of task ':build-tools:reaper:missingJavadoc'.
 Could not resolve all task dependencies for configuration ':build-tools:reaper:missingdoclet'.
    Cannot resolve external dependency org.jboss.spec.javax.annotation:jboss-annotations-api_1.2_spec:1.0.2.Final because no repositories are defined.
     Required by:
         project :build-tools:reaper > project :missing-doclet
```

I am getting this error.  Do you know how to fix this?

Reply:
I pushed a new diff with a fix.